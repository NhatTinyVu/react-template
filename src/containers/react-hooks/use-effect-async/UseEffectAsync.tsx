import React, { useState } from 'react';
import TestComponent from './TestComponent';
import { useToggleMountHandler } from './UseEffectAsync.hooks';

const SafePromise: React.FC = () => {
  const [mount, setMount] = useState<boolean>(true);
  const handleToggleMount = useToggleMountHandler({ setMount });
  return (
    <div>
      <button onClick={handleToggleMount}>Mount: {mount.toString()}</button>
      {mount && <TestComponent />}
    </div>
  );
};

export default SafePromise;
