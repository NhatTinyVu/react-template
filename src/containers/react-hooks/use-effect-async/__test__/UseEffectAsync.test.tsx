import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import UseEffectAsync from '..';

describe('EventListener', () => {
  test('renders correctly', (): void => {
    const wrapper = shallow(<UseEffectAsync />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
