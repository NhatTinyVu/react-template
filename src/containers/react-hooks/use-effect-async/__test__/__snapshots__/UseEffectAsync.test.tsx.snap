// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListener renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Mount: 
    true
  </button>
  <TestComponent />
</div>
`;
